### ISSUE TITLE
PC Relate validation suite is failing

### ISSUE THREAD

**Issue by tomwhite** [pystatgen/sgkit#374]
E.g. https://github.com/pystatgen/sgkit/runs/1355826443?check_suite_focus=true

**Comment by ravwojdyla** [comment]
Looks like something in the docker image got updated and is causing different results. Will look into this.

**Comment by ravwojdyla** [comment]
Some context on the issue:
* when I run the validation on an old image, everything works fine, thus there has to be an issue in the docker itself (likely R packages)
* BioManager has a special instructions to install specific versions of packages, documented [here](https://cran.r-project.org/web/packages/BiocManager/vignettes/BiocManager.html).

When I compare the new image with an old one, there are some updated packages, noticeably `GENESIS`: `2.18.0` -> `2.20.0` (which is where PR Relate comes from).

Kinship coefficient distributions:
![newplot (20)](https://user-images.githubusercontent.com/1419010/99241884-bb26f100-27fe-11eb-99a5-de0ff0826153.png)

**Comment by ravwojdyla** [comment]
#382 was merged, I've manually triggered the validation and it was [successful](https://github.com/pystatgen/sgkit/actions/runs/365818796).

**Comment by ravwojdyla** [comment]
We could technically close this issue, since the validation suite isn't failing anymore. But at the same time there seem to be a difference in the results, which could be interesting to investigate further. In the past we have agreed to have a point in time validation. In #382 I also add a documentation to the `pc_relate` that the user can expect results compatible with GENESIS 2.18.0. This situation might appear also for other methods like `regenie`. How do we want to handle this?

**Comment by hammer** [comment]
My vote would be to close this issue, since the validation suite is no longer failing, and file another issue to follow up on the failure caused by the GENESIS: 2.18.0 -> 2.20.0 update.

As discussed in the past, I don't think we should commit to validate our implementation against all future releases--that's just too much work. We could document this commitment someplace, I suppose.

**Comment by ravwojdyla** [comment]
Validation suite is failing again - for a different reason tho, we can reuse this issue I guess. Problem:

```
Traceback:
usr/local/lib/python3.8/dist-packages/dask/distributed.py:3: in <module>
    from distributed import *
usr/local/lib/python3.8/dist-packages/distributed/__init__.py:3: in <module>
    from .actor import Actor, ActorFuture
usr/local/lib/python3.8/dist-packages/distributed/actor.py:7: in <module>
    from .client import Future, default_client
usr/local/lib/python3.8/dist-packages/distributed/client.py:32: in <module>
    from dask.utils import ensure_dict, format_bytes, funcname, stringify
E   ImportError: cannot import name 'stringify' from 'dask.utils' (/usr/local/lib/python3.8/dist-packages/dask/utils.py)

The above exception was the direct cause of the following exception:
usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
validate_pc_relate.py:8: in <module>
    from sgkit import pc_relate
code/sgkit/__init__.py:24: in <module>
    from .stats.pca import pca
code/sgkit/stats/pca.py:6: in <module>
    from dask_ml.decomposition import TruncatedSVD
usr/local/lib/python3.8/dist-packages/dask_ml/__init__.py:4: in <module>
    from dask_ml.model_selection import _normalize
usr/local/lib/python3.8/dist-packages/dask_ml/model_selection/__init__.py:6: in <module>
    from ._hyperband import HyperbandSearchCV
usr/local/lib/python3.8/dist-packages/dask_ml/model_selection/_hyperband.py:12: in <module>
    from ._incremental import BaseIncrementalSearchCV
usr/local/lib/python3.8/dist-packages/dask_ml/model_selection/_incremental.py:19: in <module>
    from dask.distributed import Future, default_client, futures_of, wait
usr/local/lib/python3.8/dist-packages/dask/distributed.py:11: in <module>
    raise ImportError(msg) from e
E   ImportError: Dask's distributed scheduler is not installed.
E   
E   Please either conda or pip install dask distributed:
E   
E     conda install dask distributed          # either conda install
E     python -m pip install "dask[distributed]" --upgrade  # or python -m pip install
=========================== short test summary info ============================
ERROR validate_pc_relate.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 3.28s ===============================
Error: Process completed with exit code 2.
```

I haven't looked into this at all yet, but it seems like maybe the recent changes in the PCA now require `dask[distributed]`, which would be an interesting change.

**Comment by eric-czech** [comment]
It looks like distributed was always required based on checking all the changes along that trace in dask-ml.  i.e. I don't see how this could have been working without distributed before if it was still loading our PCA code.

The issue now seems to be that newer versions of distributed expect changes that don't exist in dask until 2020.12.0, specifically https://github.com/dask/dask/pull/6852, and we're pinned to 2.30.0 https://github.com/pystatgen/sgkit/pull/426.

**Comment by ravwojdyla** [comment]
Closed via #443
